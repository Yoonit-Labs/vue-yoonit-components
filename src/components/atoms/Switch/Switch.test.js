/**
 * ██╗   ██╗ ██████╗  ██████╗ ███╗   ██╗██╗████████╗
 * ╚██╗ ██╔╝██╔═══██╗██╔═══██╗████╗  ██║██║╚══██╔══╝
 *  ╚████╔╝ ██║   ██║██║   ██║██╔██╗ ██║██║   ██║
 *   ╚██╔╝  ██║   ██║██║   ██║██║╚██╗██║██║   ██║
 *    ██║   ╚██████╔╝╚██████╔╝██║ ╚████║██║   ██║
 *    ╚═╝    ╚═════╝  ╚═════╝ ╚═╝  ╚═══╝╚═╝   ╚═╝
 *
 * https://yoonit.dev - about@yoonit.dev
 *
 * Vue Yoonit Components
 * VueJS Atomic Design System framework
 *
 * Vitória Costa, Tiago Brito, Fernando Junior, Sabrina Sampaio, Gabriel Mule, Gabriel Moraes, Gabriel Rizzo & Luigui Delyer @ 2020-2021
 */

import { shallowMount } from '@vue/test-utils'
import YooSwitch from '@/components/atoms/Switch/Switch.vue'
import PropsConfig from './Switch.config'
const classBlock = 'yoo-switch'
const SlotText = 'Default Slot Text'
const mountComponent = () => {
  return shallowMount(YooSwitch, {
    slots: { default: SlotText }
  })
}
describe('YooSwitch Component', () => {
  let wrapper
  beforeEach(() => {
    jest.spyOn(global.Math, 'random').mockReturnValue(0.123456789)
    wrapper = mountComponent()
    wrapper.vm.$nextTick()
  })
  afterEach(() => {
    jest.spyOn(global.Math, 'random').mockRestore()
  })
  it('Matches Snapshot', async () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('Loads the Component HTML', () => {
    expect(wrapper.classes('yoo-switch-container')).toBe(true)
    expect(wrapper.find('input.yoo-switch').exists()).toBe(true)
  })
  describe('Props', () => {
    describe('size', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.size.options.includes(YooSwitch.props.size.default)).toBe(true)
      })
      PropsConfig.size.options.forEach(size => {
        it(`Includes size class: .${classBlock}--${size}`, async () => {
          await wrapper.setProps({ size })
          expect(wrapper.find(`.${classBlock}--${size}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props
  describe('Events', () => {
    describe('doCheck', () => {
      it('Emits doCheck Event', async () => {
        wrapper.find('.yoo-switch').trigger('click')
        expect(wrapper.emitted()).toHaveProperty('response')
      })
    })
    describe('Watchers - initialValue', () => {
      it('Check Watch initialValue', async () => {
        await wrapper.vm.$options.watch.initialValue.call(wrapper.vm, true)
        expect(wrapper.vm.checkedField).toBe(true)
      })
    })
  }) // describe Events
})
