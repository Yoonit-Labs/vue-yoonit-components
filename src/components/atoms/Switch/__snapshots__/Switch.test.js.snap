// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooSwitch Component Matches Snapshot 1`] = `<div class="yoo-switch-container m__t--s m__b--s m__r--s m__l--s"><input type="checkbox" class="yoo-switch yoo-switch--medium" id="ycid-3456789"><label class="yoo-switch__label" for="ycid-3456789"></label></div>`;
